Bowling.java

```diff
@@ -154,21 +154,44 @@ private Boolean isTurkey() {
 	private Boolean isSingleStrike() {
 		return secondToLastFrame.isPresent() && 
 		       secondToLastFrame.get().isStrike() && 
-		       !currentFrame.isStrike();
+		       !currentFrame.isStrike() &&
+					 (!thirdToLastFrame.isPresent() ||
+					   thirdToLastFrame.isPresent() && !thirdToLastFrame.get().isStrike());
 	}
```

Comment:
Ok -- let me see if I followed your thinking. You had to modify `isSingleStrike()` here so the 4th condition of the if/else statement in `updateTotalPoints()` would actually be executed when two strikes are bowled. Is this correct?

If this is the case and if you like the extracted methods, what if we switched the 3rd and the 4th conditions in `updateTotalPoints()` so that the method checks in the following order:
1) no-spare/no-strike
2) spare
3) turkey - 3 strikes
4) double - 2 strikes 
5) single - 1 strike
...otherwise, `totalPoints` doesn't get updated and we "wait" for the next bowl.

The advantage of checking in this order would be that `isSingleStrike()` doesn't not need the modification you weren't liking and we can extract the condition logic out to a method `isDoubleStrike()`. This method could even call `isSingleStrike()` I believe.

On the flip side, I agree the logic you have below is not as easily readable, but it definitely makes sense. 
Question: do you have an opinion about striving for flatter if/else statements? Or is it OK to structure your code with numerous indents?